### ISSUE TITLE
Avoid the double forward pass for online learning (TB and DB)

### ISSUE THREAD

**Issue by saleml** [GFNOrg/torchgfn#12]
A design choice was to separate the sampling of the learning objects (trajectories or transitions) and the actual learning, to allow off-policy learning. This means that two forward passes are required when doing online learning: the first pass to sample the actions to get to the next states, and the second pass to get the logits of the chosen actions. If a training object is discarded as soon as it is used for learning (i.e. online, i.e. no replay buffer), then this is inefficient, as the logits can be evaluated during the first pass. What do you think is the best way to store the "online logits" in the containers to avoid a second pass when these online logits are available ?
